### ISSUE TITLE
SDXL previewing

### ISSUE THREAD

**Issue by thundercarrot** [madebyollin/taesd#31]
I've got good previewing results with SD 1.5 and taesd and now trying SDXL and taesdxl but not getting results I expect. The first several iterations don't look like the sort of previews I was getting with SD 1.5. Here is an example with SDXL:

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/374f49a0-ec3b-4f2a-834f-4ac42914eac4" />

Several iterations later it starts to approach the final result.

The code is almost the same; the only difference is that the timestamp passed in seemed to be float type.

```
import torch
from diffusers import StableDiffusionXLPipeline, AutoencoderKL

model = "stabilityai/stable-diffusion-xl-base-1.0"
vae = AutoencoderKL.from_pretrained("madebyollin/sdxl-vae-fp16-fix", torch_dtype=torch.float16)
pipe = StableDiffusionXLPipeline.from_pretrained(model, vae=vae, torch_dtype=torch.float16)
pipe.enable_xformers_memory_efficient_attention()
pipe.enable_model_cpu_offload()

from IPython.display import display
from diffusers import AutoencoderTiny

device = torch.device("cuda")
tinyVae = AutoencoderTiny.from_pretrained("madebyollin/taesdxl", torch_dtype=torch.float16).to(device)

def get_pred_original_sample(sched, model_output, timestep, sample):
    timestep = timestep.cpu().to(torch.int64) # with SDXL timestamp appears to be a float on the cuda device
    alpha_prod_t = sched.alphas_cumprod[timestep]
    print(alpha_prod_t)
    return (sample - (1 - alpha_prod_t) ** 0.5 * model_output) / alpha_prod_t ** 0.5

preview_handle = None

def add_taesd_previewing(pipe):
    sched = pipe.scheduler
    if not hasattr(sched, "_step"):
        sched._step = sched.step
    @torch.no_grad()
    def step_and_preview(*args, **kwargs):
        global preview_images, preview_handle
        latents = get_pred_original_sample(sched, *args)
        d = tinyVae.decode(latents)[0]
        decoded = pipe.image_processor.postprocess(d)[0]
        if preview_handle is None:
            preview_handle = display(decoded, display_id=True)
        else:
            preview_handle.update(decoded)
        return sched._step(*args, **kwargs)
    sched.step = step_and_preview

add_taesd_previewing(pipe)
output = pipe(prompt="sunlit forest, photorealistic").images[0]

if preview_handle is not None:
    preview_handle.update(output)
```

What am I missing? I see that the default scheduler for SDXL differs from that of SD 1.5. That shouldn't require a change in the math?

**Comment by thundercarrot** [comment]
Perfect, thanks!
